src/api/spec/controllers/staging_project/staged_requests_controller_spec.rb

```diff
@@ -33,4 +36,62 @@
       end
     end
   end
+
+  describe 'POST #create' do
+    context 'invalid user' do
+      before do
+        login other_user
+        post :create, params: { staging_project: staging_project.name, format: :xml },
+                      body: "<requests><number>1</number><number>#{bs_request.number}</number></requests>"
+      end
+
+      it { expect(response).to have_http_status(:forbidden) }
+    end
+
+    context 'non-existent staging project' do
+      before do
+        login user
+        post :create, params: { staging_project: 'FooBar', format: :xml },
+                      body: "<requests><number>1</number><number>#{bs_request.number}</number></requests>"
+      end
+
+      it { expect(response).to have_http_status(:not_found) }
+    end
+
+    context 'with valid staging_project' do
+      before do
+        login user
+        post :create, params: { staging_project: staging_project.name, format: :xml },
+                      body: "<requests><number>1</number><number>#{bs_request.number}</number></requests>"
+      end
+
+      it { expect(response).to have_http_status(:success) }
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    context 'invalid user' do
+      before do
+        bs_request.staging_project = staging_project
+        bs_request.save
+        login other_user
+        delete :destroy, params: { staging_project: staging_project.name, format: :xml },
+                      body: "<requests><number>1</number><number>#{bs_request.number}</number></requests>"
+      end
+
+      it { expect(response).to have_http_status(:forbidden) }
+    end
+
+    context 'valid staging project and valid user' do
+      before do
+        bs_request.staging_project = staging_project
+        bs_request.save
+        login user
+        delete :destroy, params: { staging_project: staging_project.name, format: :xml },
+                      body: "<requests><number>1</number><number>#{bs_request.number}</number></requests>"
+      end
+
+      it { expect(response).to have_http_status(:success) }
```

Comment:
Should we maybe also check that the request is not staged anymore? Something like

```
it { expect(staging_project.staged_requests).not_to contain(bs_request) }
```
(untested)